File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {}
        i, j = -1, -1
        for index in range(len(nums)):
            if target - nums[index] in hashmap:
                i = hashmap[target - nums[index]]
                j = index
            hashmap[nums[index]] = index
        return [i, j]
        

File: Data Structures & Algorithms/subsets-ii/submission-1.cpp
class Solution {
public:
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        vector<vector<int>> ans;
        vector<int> path;

        sort(nums.begin(), nums.end());

        function<void(int)> backtrack = [&](int index) {
            ans.push_back(path);
            if (index == nums.size()) {
                return ;
            }

            for (int i = index; i < nums.size(); i++) {
                if (i > index && nums[i] == nums[i - 1]) continue;
                path.push_back(nums[i]);
                backtrack(i + 1);
                path.pop_back();
            }
            
        };

        backtrack(0);
        return ans;
    }
};
